patroni/dcs.py

```diff
@@ -122,8 +122,8 @@ def __init__(self, name, config):
             i.e.: `zookeeper` for zookeeper, `etcd` for etcd, etc...
         """
         self._name = name
-        self._scope = config['scope']
-        self._base_path = '/service/' + self._scope
+        self._namespace = '/{}'.format(config.get('namespace', '/service/').strip('/'))
+        self._base_path = '/'.join([self._namespace, config['scope']])
```

Comment:
I'd propose to check the isinstance for the string type and exit right away if it is not True . Because on python2 a string can be either str or unicode, and there is no unicode type in python3, I'd suggest using six, since we already require it:

    import six
    ...
    if not isinstance(self._scope, six.string_types) and not instance(self._scope, six.text_type):
        logger.fatal('Scope should be an integer, supplied {}'.format(self._scope))
        sys.exit(1)